File: docs/breez-sdk/snippets/flutter/lib/user_settings.dart
import 'package:breez_sdk_spark_flutter/breez_sdk_spark.dart';

Future<void> getUserSettings(BreezSdk sdk) async {
  // ANCHOR: get-user-settings
  final userSettings = await sdk.getUserSettings();
  print('User settings: $userSettings');
  // ANCHOR_END: get-user-settings
}

Future<void> updateUserSettings(BreezSdk sdk) async {
  // ANCHOR: update-user-settings
  await sdk.updateUserSettings(
      request: UpdateUserSettingsRequest(
          sparkPrivateModeEnabled: true,
          stableBalanceActiveLabel: null,
          sparkMasterIdentityPublicKey: SparkMasterIdentityPublicKey_Set(
              publicKey: "<hex encoded public key>")));
  // ANCHOR_END: update-user-settings
}

Future<void> activateStableBalance(BreezSdk sdk) async {
  // ANCHOR: activate-stable-balance
  await sdk.updateUserSettings(
      request: UpdateUserSettingsRequest(
          stableBalanceActiveLabel: StableBalanceActiveLabel_Set(label: "USDB")));
  // ANCHOR_END: activate-stable-balance
}

Future<void> deactivateStableBalance(BreezSdk sdk) async {
  // ANCHOR: deactivate-stable-balance
  await sdk.updateUserSettings(
      request: UpdateUserSettingsRequest(
          stableBalanceActiveLabel: StableBalanceActiveLabel_Unset()));
  // ANCHOR_END: deactivate-stable-balance
}

File: docs/breez-sdk/snippets/wasm/user_settings.ts
import { type BreezSdk } from '@breeztech/breez-sdk-spark'

const exampleGetUserSettings = async (sdk: BreezSdk) => {
  // ANCHOR: get-user-settings
  const userSettings = await sdk.getUserSettings()
  console.log(`User settings: ${JSON.stringify(userSettings)}`)
  // ANCHOR_END: get-user-settings
}

const exampleUpdateUserSettings = async (sdk: BreezSdk) => {
  // ANCHOR: update-user-settings
  await sdk.updateUserSettings({
    sparkPrivateModeEnabled: true,
    stableBalanceActiveLabel: undefined,
    sparkMasterIdentityPublicKey: {
      type: 'set',
      publicKey: '<hex encoded public key>'
    }
  })
  // ANCHOR_END: update-user-settings
}

const exampleActivateStableBalance = async (sdk: BreezSdk) => {
  // ANCHOR: activate-stable-balance
  await sdk.updateUserSettings({
    stableBalanceActiveLabel: { type: 'set', label: 'USDB' }
  })
  // ANCHOR_END: activate-stable-balance
}

const exampleDeactivateStableBalance = async (sdk: BreezSdk) => {
  // ANCHOR: deactivate-stable-balance
  await sdk.updateUserSettings({
    stableBalanceActiveLabel: { type: 'unset' }
  })
  // ANCHOR_END: deactivate-stable-balance
}
